x/lockup/abci.go

```diff
@@ -19,6 +19,14 @@ func EndBlocker(ctx sdk.Context, k keeper.Keeper) []abci.ValidatorUpdate {
 	if ctx.BlockHeight() < MinBlockHeightToBeginAutoWithdrawing {
 		return []abci.ValidatorUpdate{}
 	}
+
+	// delete synthetic locks matured before lockup deletion
+	k.DeleteAllMaturedSyntheticLocks(ctx)
+
+	// withdraw and delete locks
 	k.WithdrawAllMaturedLocks(ctx)
 	return []abci.ValidatorUpdate{}
 }
+
+// TODO: add invariant that no native lockup existent synthetic lockup exists by calling GetAllSyntheticLockups
```

Comment:
Rather than panic for this case, preventing deletion of native lockups with synthetic lockup would be better and let the tokens on lockup freeze.